test/controllers/store_controller_test.rb

```diff
@@ -6,4 +6,13 @@ class StoreControllerTest < ActionDispatch::IntegrationTest
     assert_response :success
   end
 
+  test 'index returns formatted' do
+    get store_index_url
+
+    assert_select 'nav.side_nav a', minimum: 4
+    assert_select 'main div.Polaris-Card', 3
```

Comment:
Quick suggestion - to make this more dynamic, within the test you could count how many products are in the fixture and then use this number to check how many Polaris Product Cards are being created - it would ensure that all products are being displayed on the page, rather than exactly 3 (which may break when you add more products to the fixture?)

Reply:
Ahhhhh excellent suggestion! I'll put that in thanks :)